ts/components/ButtonGroup.svelte

```diff
@@ -22,7 +31,15 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
 </script>
 
 <div {id} class="button-group btn-group {className}" {style} dir="ltr" role="group">
-    <slot />
+    <DynamicallySlottable
+        slotHost={ButtonGroupItem}
+        {createProps}
+        {updatePropsList}
+        {setSlotHostContext}
+        {api}
+    >
+        <slot />
+    </DynamicallySlottable>
```

Comment:
`DynamicallySlottable`, or rather its predecessor, used to be within ButtonGroup, but I moved it outside.

DynamicallySlottable only makes sense if it is used with ButtonGroupItem (or Item), and is supposed to allow add-ons devs to, well, dynamically slot components into it. I believe this could be a very powerful method of writing anki add-ons. Even features like the symbols insertion overlay I just implemented, uses APIs which might as well be used by add-ons.

**Because** it only makes sense if used with ButtonGroupItem, I would like them only ever to be used together, in the same component, otherwise they might get out of sync. 

This add-on API is still experimental however, mostly because it probably goes over the head of most (all ^^;) Anki add-on devs.

Reply:
Okay, thanks for the explanation. I thought I'd reduce some overhead by moving it into `ButtonGroup`  because I need `DynamicallySlottable` everytime, otherwise I get this error:
```
ButtonGroupItem: should always have a slotHostContext
```